scripts/apps/authoring-bridge/widgets/ai-widget/ai-assistant.tsx

```diff
@@ -0,0 +1,239 @@
+import React from 'react';
+import {IArticleSideWidget, IExtensionActivationResult} from 'superdesk-api';
+import {
+    Button,
+    ContentDivider,
+    Heading,
+    IconButton,
+    IllustrationButton,
+    Spacer,
+    SvgIconIllustration,
+} from 'superdesk-ui-framework/react';
+import {gettext} from 'core/utils';
+import {AuthoringWidgetHeading} from 'apps/dashboard/widget-heading';
+import {AuthoringWidgetLayout} from 'apps/dashboard/widget-layout';
+import HeadlinesTab from './headlines';
+import SummaryTab from './summary';
+import {extensions} from 'appConfig';
+
+// Can't call `gettext` in the top level
+const getLabel = () => gettext('AI Assistant');
+
+type IProps = React.ComponentProps<
+    IExtensionActivationResult['contributions']['authoringSideWidgets'][0]['component']
+>;
+
+interface IState {
+    headlinesOpen: boolean;
+    summaryOpen: boolean;
+
+    /**
+     * Handle loading of each request separately,
+     */
+    loadingHeadlines: boolean;
+    loadingSummary: boolean;
+
+    headlines: Array<string>;
+    error: boolean;
+    summary: string;
+}
+
+class AiAssistantWidget extends React.PureComponent<IProps, IState> {
+    aiAssistant: IExtensionActivationResult['contributions']['aiAssistant'];
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.aiAssistant = extensions['ai-assistant'].activationResult?.contributions?.aiAssistant;
+
+        this.state = {
+            headlinesOpen: false,
+            summaryOpen: false,
+            headlines: [],
+            error: false,
+            loadingSummary: true,
+            loadingHeadlines: true,
+            summary: '',
+        };
+
+        this.setError = this.setError.bind(this);
+        this.generateHeadlines = this.generateHeadlines.bind(this);
+        this.generateSummary = this.generateSummary.bind(this);
+    }
+
+    setError() {
+        this.setState({
+            error: true,
+        });
+    }
+
+    generateHeadlines() {
+        if (this.aiAssistant?.generateHeadlines == null) {
```

Comment:
fyi, you commented on the previous implementation, but the new one is similar. Well, that can't be done with the design you suggested. Using this design it's possible to register the extension, but you can skip registering the custom API requests. So that's why if someone wants, they can register the extension. And actually for the future I see a benefit since since some clients might want to have e.g. summary, translations but not headlines generation.

Reply:
In another comment I asked to make `generateHeadlines/generateSummary` mandatory. If you want to keep it optional, update the UI so if headlines/summary is not passed - buttons to generate shouldn't appear either.

And if buttons appear only when it's registered - there's no use in this function to again check if it's registered.